### PR TITLE
Fix Autocompleter promises

### PR DESCRIPTION
**Pull request by lukebarnard1**
Use bluebird instead of the now removed "q" library.

Also, make sure we timeout and then `reflect` to effectively do an `allSettled` that waits for all promises to either be resolved or rejected. Then we filter for those that are fulfilled and return the completions.

### CODE REVIEW

File: src/autocomplete/Autocompleter.js [hunk 1]
```diff
@@ -52,28 +52,30 @@ export async function getCompletions(query: string, selection: SelectionRange, f
...
+    */
+    const completionsList = await Promise.all(
+        // Array of inspections of promises that might timeout. Instead of allowing a
+        // single timeout to reject the Promise.all, reflect each one and once they've all
+        // settled, filter for the fulfilled ones
+        PROVIDERS.map((provider) => {
+            // Convert to bluebird promise so that we can do a timeout
+            const p = Promise.resolve(provider.getCompletions(query, selection, force));
+            return p.timeout(PROVIDER_COMPLETION_TIMEOUT);
+        }).map((p) => p.reflect()),
```
Comment (richvdh) [inline src/autocomplete/Autocompleter.js]: do you need a separate `map` here? why not just tag the `reflect` onto the `p.timeout` line?

File: src/autocomplete/Autocompleter.js [hunk 2]
```diff
@@ -52,28 +52,30 @@ export async function getCompletions(query: string, selection: SelectionRange, f
...
-            return Promise.resolve(provider.getCompletions(query, selection, force))
-                .timeout(PROVIDER_COMPLETION_TIMEOUT);
-        }),
+    */
+    const completionsList = await Promise.all(
+        // Array of inspections of promises that might timeout. Instead of allowing a
+        // single timeout to reject the Promise.all, reflect each one and once they've all
+        // settled, filter for the fulfilled ones
+        PROVIDERS.map((provider) => {
+            // Convert to bluebird promise so that we can do a timeout
```
Comment (richvdh) [inline src/autocomplete/Autocompleter.js]: hrm; I think we've just decided to do the opposite in #riot-dev

Comment (lukebarnard1) [inline src/autocomplete/Autocompleter.js]: That's ok, this will still work.

File: src/autocomplete/Autocompleter.js (hunk 1 above)
Comment (lukebarnard1) [inline src/autocomplete/Autocompleter.js]: Fair. Done.

**Pull request merged**